### ISSUE TITLE
Improve UI for importing a backup job where the backup files have been protected with encryption passphrase

### ISSUE THREAD

**Issue by luixxiul** [duplicati/duplicati#5499]
**Is your feature request related to a problem? Please describe.**

When importing a file for backup job on another computer than the original one in order to restore the backups by repairing the database of the job, I have struggled to restore them with the file, since I have forgotten whether backups were encrypted or not with a password (AES-256).

Having read https://forum.duplicati.com/t/set-backup-back/2485/4, I tried a password which I vaguely remember may be related to Duplicati, which was rejected explicitly with an error notification, then I tried empty password, which I also have set to other jobs, and it failed too because of another error during repairing the database.

I failed to save the error log, but it did not seem to be related to the password at a glance, instead I remember the error contained a lot of `Error on Backup: InvalidManifestException: No manifest file found in volume` (I may be wrong here), so I was not able to understand why the database was not successfully repaired.

Checking the bucket directly, it turned out that the backup files were protected with another valid password, and providing it made it possible to repair the backups and its job successfully.

**Describe the solution you'd like**

If backups are protected with a password, running repair with an empty password should also display an error notification which contains a concise message that the backups are protected with a password, asking users to provide one.

Asking users to open a error log page which contains just technical information is not really user-friendly and can have them rather feel lost, if they are not tech savvy.

**Describe alternatives you've considered**

**Additional context**

The error message here should be very clear and explicit, so that users can feel safe and easily understand what to do, since the errors experienced during restore are indeed stressful and not good for Quality of Life, taking the fact that notification messages cannot be translated [just yet](https://github.com/duplicati/duplicati/issues/5305#ref-commit-bb7b128) into consideration.

**Comment by ts678** [comment]
Is this GUI or CLI? GUI won't let you save a job with an empty password, if encryption is on, and it's on if it was on at export.

![image](https://github.com/user-attachments/assets/2bb35b83-a365-4b2a-9896-ebf7c97ac141)

![image](https://github.com/user-attachments/assets/a2f9984e-05d7-4ae9-88f0-75b74223599b)

Can you provide exact steps to reproduce this, starting with Export, where you have to uncheck a box to omit the password:

![image](https://github.com/user-attachments/assets/96b8ed5d-1d28-431c-a0af-61d083a8e236)

**Comment by luixxiul** [comment]
Of course. Here is the complete steps to reproduce the issue from a scratch:

1. Remove knowledge about usage of Duplicati and be a novice user, who is not tech savvy
1. Start Debian on your machine, if your OS is not one
2. Install and start Duplicati
3. Open Firefox
4. Go to http://localhost:8200/ngax/index.html#/addstart 
5. Click "Configure a new backup"
6. Click "Next >"
7. Click the input area "Name"
8. Input "Test"
9. Click "Generate" on the bottom, under "Repeat Passphrase"
1. Take note of the passphrase
1. Click "Next >"
1. Click "Manually type path"
1. Click the input area "Folder path"
1. Input "/tmp"
1. Click "Next >"
1. Click "Home" in "Source data" area
1. Click "Next >"
1. Click "Next >" again
1. Click "Save"
1. Click "Yes, I have stored the passphrase safely"
1. Click "Run now"
1. Wait until the backup is completed
1. Click "Test"
1. Click "Export …"
1. Uncheck "Export passwords"
1. Click "Export"
1. Save the file by clicking "OK" and "Save"
1. Click "Done"
1. Click "Delete …"
1. Click "Delete backup >"
1. Click "Yes"
1. Wait until you will forget whether you have set a passphrase to protect the backups, but before /tmp directory is eventually emptied :smirk: 
1. Then, go to http://localhost:8200/ngax/index.html#/addstart
1. Select "Import from a file"
1. Click "Next >"
1. Click "Browse …" next to "Configuration file:"
1. Find and select the backup job file, which should be named as "Test-duplicati-config.json"
1. Click "Import"
1. Click the pulldown next to "Encryption"
1. Select "No encryption"
1. Not be worried with the message `We recommend that you encrypt all backups stored outside your system` as you are sure that the backup exists locally
1. Click "Manually type path"
1. Click the input area "Folder path"
1. Input "/tmp"
1. Click "Next >"
1. Make sure "Home" is selected
1. Click "Next >"
1. Click "Next >" again
1. Click "Save"
1. Click "Test"
1. Click "Database …"
1. Click "Repair"
1. Make sure the notification banner titled with "Error while running Test" is displayed with "Got (x) error(s)" where x depends on the number of backed up files:
![1](https://github.com/user-attachments/assets/ba852950-9908-40a5-97b2-4d833e6254c7)
1. Click "Show"
1. Click the item to expand it
1. Click "Errors"
1. Make sure the multiple errors "InvalidManifestException: No manifest file found in volume" are displayed
![2](https://github.com/user-attachments/assets/d43275b9-9f0f-4057-9e48-bfbfc9f7ae3d)
1. Be confused and worried :frowning: 
1. Then, remember you have set a passphrase in order to protect the backups :astonished: 
1. Click "Dismiss"
1. Click "Home"
1. Click "Test"
1. Click "Delete …"
1. Click "Delete backup >"
1. Click "Yes" to start over
1. Go to http://localhost:8200/ngax/index.html#/addstart
1. Select "Import from a file"
1. Click "Next >"
1. Click "Browse …" next to "Configuration file:"
1. Select "Test-duplicati-config.json"
1. Click "Import"
1. Input the passphrase you have set on the step 10 to the input area "Passphrase"
1. Input the same passphrase to the input area "Repeat Passphrase"
1. Click "Options"
1. Click "Save"
1. Click "Test"
1. Click "Database …"
1. Click "Repair"
1. Wait until the database is repaired
1. Click "Test"
1. Click "Restore files …"
1. Be relieved at the directories being displayed :slightly_smiling_face: 
![3](https://github.com/user-attachments/assets/eb368da2-65de-4417-816e-95aaecaf439d)
1. Click ".bash_history" for example
1. Click "Continue"
1. Click "Pick location"
1. Click "Browse"
1. Click "Home"
1. Click "Downloads" for example
1. Click "Restore"
1. Open the directory "Downloads"
1. Make sure the file ".bash_history" exists and can be opened
1. Be relieved again at the fact that the backup was not gone :smile:

**Expected result:**

There should be a friendly reminder or indication that the backup seems to be protected with a passphrase, if it is indeed protected with one but the user does not provide any.

**Actual result:**

Leading the user to the page where he/she will find a bunch of errors `InvalidManifestException: No manifest file found in volume` like above on step 58 is not really friendly, if the user is not tech savvy and therefore easily be confused with such technical information.

Also, let's not forget the plausible fact that GUI users prefer it over command line because they are either not familiar with command line or not like it, and displaying error messages which you usually see on the command line does not really help anyone.

**Comment by ts678** [comment]
Thanks for clarifying. What threw me off was "empty password". Steps actually turn encryption off.

> indication that the backup seems to be protected with a passphrase

The [indication](https://github.com/duplicati/duplicati/issues/5499#issuecomment-2316121491) was in the GUI, with encryption turned on, and two blank lines wanting passphrase.
Novice users wouldn't know that, and creating new job from scratch looks the same as the above.

> Save a flag in the exported backup file that it is supposed to be encrypted

If I import rather than "new backup", GUI currently seems to come up with past encryption setting.

> Give a better error message when attempting to open an encrypted volume without a password

Yes, the current one is obscure to anyone who isn't deeply into internals. There are more like that...

**Comment by kenkendk** [comment]
@luixxiul I welcome all suggestions for improving the usability, and I don't view this as nitpicking. If we can steer the user in the right direction we should.

**Comment by ts678** [comment]
There's an ordering problem to consider. Screen 1 is where the encryption is set. Destination info is set on screen 2.
Manual entry would probably go in screen order. Import is probably all at once. Direct restore sets destination first.
This all has to do with whether Duplicati actually looks at destination for signs of encryption, or uses imported info.

**Comment by kenkendk** [comment]
This is fixed in ngclient.

The new API call figures out that encrypted files are present and returns an suitable error message if no passphrase was given.
